cmd/lifecycle/main.go

```diff
@@ -104,3 +104,12 @@ func initCache(cacheImageTag, cacheDir string, keychain authn.Keychain) (lifecyc
 	}
 	return cacheStore, nil
 }
+
+func appendNotEmpty(slice []string, elems ...string) []string {
+	for _, v := range elems {
+		if v != "" {
+			slice = append(slice, v)
```

Comment:
When calling `append`, if `slice` array is too small to fit all the given values, a bigger array will be allocated. Due to this performance issue, theoretically it would be better to know how many elements we're going to append to `slice`, increase its size (if needed) and then `append` the elements.
Currently we're appending a small number of elements on each call so I'm not sure it's worth such change, but I think we should at least keep it in mind.